### ISSUE TITLE
Probes: replace bpftool kconfig probes with a pure Go API

### ISSUE THREAD

**Issue by rgo3** [cilium/cilium#20701]
Currently the ProbeManager has a function that uses `bpftool`'s output to verify certain kconfig settings.

In order to deprecate ProbeManager (#20062) and not having to shell out to `bpftool` we should investigate if we need to add an API to the `probes` pkg that parses the kconfig and checks for needed values in pure Go or if we can get away with other already existing probes (ideas mentioned in #14314).

**Comment by ti-mo** [comment]
Before we start hacking on this, we need to do some discovery around which kconfig values are actually relied upon, and whether or not this is necessary to begin with.

2 things for context:
- New libbpf has `extern kconfig`, which we're adding preliminary support for in `cilium/ebpf`: https://github.com/cilium/ebpf/issues/698. This is for accessing kconfig values from BPF C, but the libbpf implementation is not compatible with Cilium's static data implementation.
- When running Cilium containerized, a kernel config might simply not be available. Debian kernels compile without `/proc/config(.gz)`, and the kconfig needs to be pulled from `/boot` on the host, which the Cilium container should never have access to.

All in all, kconfig searching/parsing/probing is unreliable and we should try to avoid relying on it if at all possible.

**Comment by ti-mo** [comment]
Superseded by https://github.com/cilium/cilium/issues/22514.
